src/main/java/org/jabref/gui/exporter/CreateModifyExporterDialogViewModel.java

```diff
@@ -0,0 +1,120 @@
+package org.jabref.gui.exporter;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.util.FileDialogConfiguration;
+import org.jabref.logic.exporter.SavePreferences;
+import org.jabref.logic.exporter.TemplateExporter;
+import org.jabref.logic.journals.JournalAbbreviationLoader;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.layout.LayoutFormatterPreferences;
+import org.jabref.logic.util.FileType;
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * This view model can be used both for "add exporter" and "modify exporter" functionalities.
+ * It takes an optional exporter which is empty for "add exporter," and takes the selected exporter
+ * for "modify exporter."  It returns an optional exporter which empty if an invalid or no exporter is
+ * created, and otherwise contains the exporter to be added or that is modified.
+ *
+ */
+
+public class CreateModifyExporterDialogViewModel extends AbstractViewModel {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateModifyExporterDialogViewModel.class);
+
+    private final DialogService dialogService;
+    private final PreferencesService preferences;
+
+    private final StringProperty name = new SimpleStringProperty("");
+    private final StringProperty layoutFile = new SimpleStringProperty("");
+    private final StringProperty extension = new SimpleStringProperty("");
+
+    private final JournalAbbreviationLoader loader;
+
+
+    public CreateModifyExporterDialogViewModel(Optional<ExporterViewModel> exporter, DialogService dialogService, PreferencesService preferences,
+                                               JournalAbbreviationLoader loader) {
+        this.dialogService = dialogService;
+        this.preferences = preferences;
+        this.loader = loader;
+
+        setTextFields(exporter);
+
+
```

Comment:
I am sure checkstyle will complain about too many empty lines(not more than one)

Reply:
I am fixing spacing and putting it in my next push.  Thanks!